tests/test_rtdc_config.py

```diff
@@ -81,6 +85,284 @@ def test_config_update():
+                                 "more metadata": True}
+    # overwrite the previous keys and valus
+    ds.config["user"] = {}
+    assert ds.config["user"] == {}
+
+
+def test_user_section_set_save_reload_empty_dict():
+    """The 'user' config section as an empty dict will not save"""
+    h5path = retrieve_data("rtdc_data_hdf5_rtfdc.zip")
+    with new_dataset(h5path) as ds:
+        ds.config.update({"user": {}})
+        expath = h5path.with_name("exported.rtdc")
+        ds.export.hdf5(expath, features=ds.features_innate)
+    # fails for hdf5
+    with pytest.raises(KeyError):
+        with h5py.File(expath, "r") as h5:
+            assert h5.attrs["user:"] == {}
+    # works for dclab because "user" added when checked
+    with new_dataset(expath) as ds2:
+        assert ds2.config["user"] == {}
+
+
+def test_user_section_set_save_reload_empty_key():
+    """Empty 'user' section key value allowed"""
+    h5path = retrieve_data("rtdc_data_hdf5_rtfdc.zip")
+    with new_dataset(h5path) as ds:
+        ds.config.update({"user": {"": " "}})
+        expath = h5path.with_name("exported.rtdc")
+        ds.export.hdf5(expath, features=ds.features_innate)
+    # make sure that worked
+    with h5py.File(expath, "r") as h5:
+        assert h5.attrs["user:"] == " "
+    # now check again with dclab
+    with new_dataset(expath) as ds2:
+        assert ds2.config["user"] == {"": " "}
+
+
+@pytest.mark.parametrize("user_config", [{"": ""}, {" ": ""}])
+def test_user_section_set_save_reload_fails(user_config):
+    """Show the empty string configurations that are not allowed"""
+    h5path = retrieve_data("rtdc_data_hdf5_rtfdc.zip")
+    with new_dataset(h5path) as ds:
+        ds.config.update({"user": user_config})
+        expath = h5path.with_name("exported.rtdc")
+        ds.export.hdf5(expath, features=ds.features_innate)
+    # make sure that "user" does not exist for an empty dict
+    with pytest.raises(KeyError):
+        with h5py.File(expath, "r") as h5:
+            assert h5.attrs["user:"] == user_config
+    with pytest.raises(AssertionError):
+        with new_dataset(expath) as ds2:
+            assert ds2.config["user"] == user_config
+
+
+def test_user_section_set_save_reload_fmt_dict():
+    """Check that 'user' section metadata works for RTDC_Dict"""
+    # create temp directory for storing outputted file
+    tpath = pathlib.Path(tempfile.mkdtemp())
+    ddict = example_data_dict(size=67, keys=["area_um", "deform"])
+    ds = new_dataset(ddict)
+    metadata = {"some metadata": 42}
+    ds.config.update({"user": metadata})
+    assert ds.config["user"] == metadata
+
+    expath = tpath / "exported.rtdc"
+    with expath as exp:
+        ds.export.hdf5(exp, features=["deform", "area_um"])
+    # make sure that worked
+    with h5py.File(expath, "r") as h5:
+        assert h5.attrs["user:some metadata"] == 42
+    # fails due to RTDC_HDF5 trying to add self.title
+    with pytest.raises(KeyError):
```

Comment:
Please change this test so that this works (no KeyError). i.e. add `ds.config["experiment"]["sample"]="test"` and `ds.config["experiment"]["run index"]=1` before exporting.

Reply:
Ah, of course...